scipy/linalg/tests/test_lapack.py

```diff
@@ -406,6 +406,19 @@ def test_gelsy(self):
                             rtol=25*np.finfo(dtype).eps)
 
 
+@pytest.mark.parametrize('dtype', DTYPES)
+def test_geqrf_lwork(dtype):
+    geqrf, geqrf_lwork = get_lapack_funcs(
+        ('geqrf', 'geqrf_lwork'),
+        dtype=dtype,
+    )
+    m, n = 4, 3
+    _, _, work, _ = geqrf(np.ones((m, n), dtype=dtype), lwork=-1)
+    lwork, info = geqrf_lwork(m=m, n=n)
+    assert_equal(info, 0)
+    assert_equal(lwork, work[0])
```

Comment:
As this is failing on Travis, you might want to `int` them as they return in the dtype of the work. In production we use `_compute_lwork` function to strip off the integer out from the work query. Then you can call `assert_equal`.